### PR TITLE
Add git hash to `distributed-impl` version

### PR DESCRIPTION
**Pull request by charlesbluca**
Makes each `distributed-impl` package unique to resolve the upload errors observed in #5864

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: continuous_integration/recipes/distributed/meta.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ build:
 
 outputs:
   - name: distributed-impl
-    version: {{ version }}
+    version: {{ version }}+{{ GIT_DESCRIBE_HASH }}
```
Comment (jakirkham) [inline continuous_integration/recipes/distributed/meta.yaml:21]: Should we add this to the other packages here as well?

Comment (jakirkham) [reply continuous_integration/recipes/distributed/meta.yaml:21]: Ah nvm we have this in the `build/string` in other cases

Comment (charlesbluca) [reply continuous_integration/recipes/distributed/meta.yaml:21]: I'm fine with that! I think that the cython builds are young enough that slightly altering the build string shouldn't cause any issues 🙂

**Comment by jakirkham** [comment]
Thanks Charles! 😀

**Pull request merged**